elastic/datadog_checks/elastic/elastic.py

```diff
@@ -398,25 +398,19 @@ def _process_metric(self, data, metric, xtype, path, xform=None, tags=None, host
             self.log.debug("Metric not found: %s -> %s", path, metric)
 
     def _process_health_data(self, data, version, base_tags, service_check_tags):
-        cluster_status = data.get('status')
-        if not self.cluster_status.get(self._config.url):
-            self.cluster_status[self._config.url] = cluster_status
-            if cluster_status in ["yellow", "red"]:
-                event = self._create_event(cluster_status, tags=base_tags)
-                self.event(event)
-
-        if cluster_status != self.cluster_status.get(self._config.url):
-            self.cluster_status[self._config.url] = cluster_status
-            event = self._create_event(cluster_status, tags=base_tags)
-            self.event(event)
-
-        cluster_health_metrics = health_stats_for_version(version)
-
-        for metric, desc in iteritems(cluster_health_metrics):
+        prev_status = self.cluster_status.get(self._config.url)
+        self.cluster_status[self._config.url] = current_status = data.get('status')
+        if self._config.submit_events and (
+            (prev_status is None and current_status in ["yellow", "red"])  # Cluster starts in bad status.
+            or current_status != prev_status
```

Comment:
Non blocking nit:
Are you purposely keeping these conditions separate for legibility? I think the `prev_status is None` seems to be a bit redundant to me. But I can understand it from a legibility to have each condition listed.

Reply:
Ah, good question! I started out with just `current_status != prev_status`, I think that's similar to what you're thinking, right?

Unfortunately, the logic is actually: "send an event if the cluster *starts* in yellow/red state OR if cluster state changes". And that's what the two conditions correspond to exactly.

If I misunderstood you and you meant to just get rid of `prev_status is None`, that would result in us submitting an event every time the cluster is in yellow/red state, which would submit a lot more events that we currently do.

My understanding of events is that (unlike service checks or metrics) we don't submit them with every check run.

Reply:
Ahh yeah your right! I had a lapse of judgment there and had to bring out the pen and paper.

Reply:
all good, I needed a pen and paper to reason through it too!